### DISCUSSION TITLE
Question about nccopy

### DISCUSSION THREAD

**Discussion by jedwards4b** [Unidata/netcdf-c#2632] [Q&A] [answered]
I'm trying to follow the flow of nccopy.c and I have a question about [this call to
count_dims](https://github.com/Unidata/netcdf-c/blob/main/ncdump/nccopy.c#L2116)
It's called with igrp which is the ncid of the file opened to read.   In count_dims we have a 
[call to nc_inq_grps](https://github.com/Unidata/netcdf-c/blob/main/ncdump/nccopy.c#L1785)
My question is why can nccopy call this when the input file is in a netcdf3 format and not get a NC_ENOTNC4  
error?

**Comment by DennisHeimbigner** [comment, marked as answer]
Where possible, we try to define the netcdfd-4 API functions for netcdf-3: see libsrc/nc3dispatch.c.
In this case, a netcdf-3 file is treated as if it has a single root group and no subgroups.

**Reply by jedwards4b** [reply]
thanks
